.github/workflows/gradle-build-with-detekt.yml

```diff
@@ -1,6 +1,7 @@
 name: Gradle Build With Detekt
 
-on: [push, pull_request]
+# Detekt doesn't work with Kotlin 2.0 for now, so it's disabled by default
+on: [workflow_dispatch]
```

Comment:
Here you have disabled the action execution on pull request or push, I'm not sure if this should be done. Especially I don't understand why **Detekt** won't run. In order to change the behavior of the action during execution, you need to change `jobs`.